shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -176,15 +179,27 @@ - (void)setViewController:(FlutterViewController*)viewController {
   self.iosPlatformView->SetOwnerViewController(_viewController);
   [self maybeSetupPlatformViewChannels];
 
+  __block FlutterEngine* blockSelf = self;
```

Comment:
🔮 The retain cycle will be reintroduced when someone finally hits the "modernize to ARC" button without auditing all the blocks where weak memory management is expected. 🔮

Reply:
New word, "retro-modernize"?  When we move to ARC it should work as well.  There is a subtle difference between __block and __weak (like how they treat primitives).  For an object pointer it should behave the same.  I'm more sure about that than my understanding of the `addObserverForName:object:queue:usingBlock:` but obviously I've been wrong before, haha.